skimage/filters/_gabor.py

```diff
@@ -72,30 +90,134 @@ def gabor_kernel(frequency, theta=0, bandwidth=1, sigma_x=None, sigma_y=None,
     >>> io.imshow(gk.real)  # doctest: +SKIP
     >>> io.show()           # doctest: +SKIP
     """
-    if sigma_x is None:
-        sigma_x = _sigma_prefactor(bandwidth) / frequency
-    if sigma_y is None:
-        sigma_y = _sigma_prefactor(bandwidth) / frequency
+    def __new__(cls, frequency, thetas=None, bandwidth=1, sigma=None,
+                sigma_y=None, n_stds=3, offset=None, ndim=2, **kwargs):
+        # Import has to be here due to circular import error
+        from ..transform import rotate
+
+        # handle deprecation
+        message = ('Using deprecated, 2D-only interface to'
+                   'gabor_kernel. This interface will be'
+                   'removed in scikit-image 0.16. Use'
+                   'gabor_kernel(frequency, rotation=theta,'
```

Comment:
You say `rotation=theta` here but the signature does not have a rotation keyword. fwiw I think theta as a kwarg can be maintained, silently adding support for multidimensional theta.